src/api/eventTypes.js

```diff
@@ -171,3 +172,118 @@ export type RestartEvent = $ReadOnly<{|
   zulip_version?: string,
   zulip_feature_level?: number,
 |}>;
+
+type RealmData = $ReadOnly<{|
+  allow_edit_history?: boolean,
+  bot_creation_policy?: number,
+  create_stream_policy?: number,
+  invite_to_stream_policy?: number,
+  move_messages_between_streams_policy?: number,
+  wildcard_mention_policy?: number,
+  description?: string,
+  realm_giphy_rating?: number,
+  digest_emails_enabled?: number,
+  disallow_disposable_email_addresses?: number,
+  email_address_visibility?: number,
+  email_changes_disabled?: boolean,
+  invite_required?: boolean,
+  inline_image_preview?: boolean,
+  inline_url_embed_preview?: boolean,
+  mandatory_topics?: boolean,
+  realm_name?: string,
+  name_changes_disabled?: boolean,
+  avatar_changes_disabled?: boolean,
+  emails_restricted_to_domains?: boolean,
+  send_welcome_emails?: boolean,
+  message_content_allowed_in_email_notifications?: boolean,
+  video_chat_provider?: number,
+  waiting_period_threshold?: number,
+  digest_weekday?: number,
+  private_message_policy?: number,
+  user_group_edit_policy?: number,
+  default_code_block_language?: string,
+  message_content_delete_limit_seconds?: number,
+
+  // Fails `value?: string | boolean | number` check on RealmUpdateEvent.
+  // authentication_methods?: {| +[authentication_method_name: string]: boolean |},
```

Comment:
It looks like all (or most?) of the properties here are meant to correspond to properties in `InitialDataRealm`.

Awkwardly we can't actually use that type, or express in the type system its relationship to this one, because the names differ: there's that `realm_` prefix on the initial-data side.

But let's use the same types for both, and where they need to be updated let's do so in tandem. We can also keep them listed in the same order, to facilitate visually comparing them (and spotting where one side has a property that the other doesn't.)

We can also make a type-test, to confirm they stay in sync for each given property. (This will be most useful for the properties like this one where the type might get updated over time.) The type-test could have a long list of lines like:
```js
  check<'realm_authentication_methods', 'authentication_methods'>();
  check<'realm_avatar_changes_disabled', 'avatar_changes_disabled'>();
  // … etc. etc. etc., one per property
```
where `check` can be written somewhat like `typesEquivalent`. I can make a go at writing that later.

Then we'll still need to rely on comments to remind us to keep them in sync when adding new properties, but at least the test will help when updating one of these complex properties' types.